processor/common/common.go

```diff
@@ -0,0 +1,340 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/spf13/cast"
+)
+
+var (
+	// TODO Add processor type invoking the BuildMatchProperties in error text.
+	errAtLeastOneMatchFieldNeeded = errors.New(
+		`error creating processor. At least one ` +
+			`of "services", "span_names" or "attributes" field must be specified"`)
+
+	errInvalidMatchType = fmt.Errorf(
+		`match_type must be either %q or %q`, MatchTypeStrict, MatchTypeRegexp)
+)
+
+// matchingProperties is an interface that allows matching a span against a configuration
+// of a match.
+type MatchingProperties interface {
```

Comment:
This is span-related functionality. We may in the future want to have similar metric-related functionality. To avoid name clashes I suggest we name this `MatchingSpanProperties` (similarly add the word Span everywhere else).
Perhaps an even better name for this interface would be `SpanPropertyMatcher`.

Reply:
Hum at first look this feels more like a `Matcher` per golang idiomatic naming practices. Let's see with the rest of the code.